### PR TITLE
bf: ZENKO-436 retrieve correct read location constraint

### PR DESCRIPTION
**Pull request by alexanderchan-scality**

### CODE REVIEW

File: tests/unit/utils/retrieveDefaultRead.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            loc: 'nobucketmatch',
+            results: {
+                location: 'nobucketmatch',
+                key: `${bucket}/${key}`,
+                locationType: 'aws_s3',
+            },
+        },
+    ];
+
+    tests.forEach(test => it(test.it, () => {
```
Comment (jonathan-gramain) [inline tests/unit/utils/retrieveDefaultRead.js]: I usually like `forEach` loops for test factorization, here it seems duplication is minimal and does not really justify the use of a loop compared to explicit separate tests (for readability and it's then easier to add custom checks to each if needed).

File: tests/unit/utils/retrieveDefaultRead.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    let config;
+
+    before(() => {
+        config = new ConfigObject();
+        config.setLocationConstraints(locationConstraints);
+    });
+
+    const tests = [
+        {
+            it: 'should return null location constraints does not exist',
```
Comment (jonathan-gramain) [inline tests/unit/utils/retrieveDefaultRead.js]: I guess you meant "... if does not exist"

File: lib/api/apiUtils/object/retrieveDefaultRead.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+
+function retrieveDefaultRead(config, locationName, objectKey, bucketName) {
```
Comment (bennettbuchanan) [inline lib/api/apiUtils/object/retrieveDefaultRead.js]: I think it's worth adding a JSDoc for this. 😄

**Comment by alexanderchan-scality** [comment]
Changed function name and filename to better describe the function and added JSDoc.
Made changes to the tests as requested

File: lib/api/apiUtils/object/checkReadLocation.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+/**
+ * checkReadLocation - verify that a bucket's default read location exists
+ * a specified read data locator
```
Comment (jonathan-gramain) [inline lib/api/apiUtils/object/checkReadLocation.js:3]: `for a specified`

**Pull request merged**